pdns/pdns_recursor.cc

```diff
@@ -1835,26 +1837,34 @@ static void startDoResolve(void *p)
       // Does the validation mode or query demand validation?
       if(!shouldNotValidate && sr.isDNSSECValidationRequested()) {
         try {
-          if(sr.doLog()) {
-            g_log<<Logger::Warning<<"Starting validation of answer to "<<dc->d_mdp.d_qname<<"|"<<QType(dc->d_mdp.d_qtype).getName()<<" for "<<dc->getRemote()<<endl;
+          auto state = sr.getValidationState();
+
+          string x_marker;
+          if (sr.doLog() || vStateIsBogus(state)) {
+            auto xdnssec = g_xdnssec.getLocal();
+            if (xdnssec->check(dc->d_mdp.d_qname)) {
+              x_marker = " [in x-dnssec-names]";
+            }
           }
 
-          auto state = sr.getValidationState();
+          if(sr.doLog()) {
+            g_log<<Logger::Warning<<"Starting validation of answer to "<<dc->d_mdp.d_qname<<"|"<<QType(dc->d_mdp.d_qtype).getName()<<x_marker<<" for "<<dc->getRemote()<<endl;
```

Comment:
I think this line should go, actually. It used to make sense in 4.0 when we were doing validation here, after everything else, but I think it does not make sense anymore.